osi_sensorviewconfiguration.proto

```diff
@@ -0,0 +1,632 @@
+
+    // Emitter Frequency
+    //
+    // This information can be used by a ray tracing engine to calculate
+    // doppler shift information and take into account differences in
+    // refraction and reflection.  For doppler shift calculations the
+    // sensor model can of course always provide a nominal frequency and
+    // adjust the resulting doppler shift information to actual frequency
+    // through frequency adjustments.  For material and geometry interaction
+    // purposes the frequency is also relevant.
+    //
+    // Unit: [Hz]
+    optional double emitter_frequency = 9;
+}
+
+//
+// \brief The configuration settings for the Camera Sensor Input to be provided
+// by the environment simulation.
+//
+message CameraSensorViewConfiguration
+{
+    // The ID of the sensor at host vehicle's mounting_position.
+    //
+    // This is the ID of the physical sensor, to be used in its detected
+    // features output; it is distinct from the ID of its virtual sensor.
+    //
+    // The ID is to be provided by the environment simulation, the sensor
+    // model is not in a position to provide a useful default value.
+    //
+    optional Identifier sensor_id = 1;
+
+    // The physical mounting position of the sensor (origin and orientation
+    // of the sensor coordinate system) given in vehicle coordinates [1].
+    // The physical position pertains to this detector individually, and
+    // governs the sensor-relative coordinates in features detected by this
+    // detector.
+    //
+    // \arg \b x-direction of sensor coordinate system: sensor viewing direction
+    // \arg \b z-direction of sensor coordinate system: sensor (up)
+    // \arg \b y-direction of sensor coordinate system: perpendicular to x and z
+    // right hand system
+    //
+    // \par References:
+    // [1] DIN ISO 8855:2013-11
+    //
+    // \note The origin of vehicle's coordinate system in world frame is
+    // ( \c Vehicle::base.position + Inverse_Rotation_yaw_pitch_roll(
+    // \c Vehicle::base.orientation) * \c Vehicle::bbcenter_to_rear ).
+    // The orientation of the vehicle's coordinate system is equal to the
+    // orientation of the vehicle's bounding box \c Vehicle::base.orientation.
+    // \note A default position can be provided by the sensor model (e.g. to
+    // indicate the position the model was validated for),
+    // but this is optional; the environment simulation must provide a valid
+    // mounting position (based on the vehicle configuration) when setting the
+    // view configuration.
+    //
+    optional MountingPosition mounting_position = 2;
+
+    // The root mean squared error of the mounting position.
+    //
+    optional MountingPosition mounting_position_rmse = 3;
+
+    // Field of View in horizontal orientation of the physical sensor
+    //
+    // Viewing range: [-field_of_view_horizontal/2, field_of_view_horizontal/2]
+    // azimuth in the sensor frame as defined in \c OSI:Spherical3d.
+    // Unit: [rad]
```

Comment:
Should we add information about distortion? Ist the field of view the min. or max. field of view (as example barrel distortion for cameras)?